src/common/Gateway.sol

```diff
@@ -29,23 +27,16 @@ import {MessageProofLib} from "src/common/libraries/MessageProofLib.sol";
 ///         and proofs that have been received. Also implements a retry method for failed messages.
 contract Gateway is Auth, Recoverable, IGateway {
     using BytesLib for bytes;
-    using MathLib for uint256;
-    using MessageProofLib for *;
-    using ArrayLib for uint16[8];
     using TransientStorageLib for bytes32;
 
-    uint8 public constant MAX_ADAPTER_COUNT = 8;
-    uint8 public constant PRIMARY_ADAPTER_ID = 1;
     PoolId public constant GLOBAL_POT = PoolId.wrap(0);
-    uint256 public constant RECOVERY_CHALLENGE_PERIOD = 7 days;
     bytes32 public constant BATCH_LOCATORS_SLOT = bytes32(uint256(keccak256("Centrifuge/batch-locators")) - 1);
 
-    uint16 public immutable localCentrifugeId;
-
     // Dependencies
     IRoot public immutable root;
     IGasService public gasService;
     IMessageProcessor public processor;
+    IAdapter public adapter;
```

Comment:
Seems to me like we are missing `file` support for the adapter as its not immutable?

Reply:
Done